packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift

```diff
@@ -601,9 +589,77 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
         result.append(startNode)
         unresolved.remove(startNode)
     }
-    
+
     // TODO: Remove once all configure is moved to the bridge
     func getPlugin() throws -> AWSDataStorePlugin {
         return try Amplify.DataStore.getPlugin(for: "awsDataStorePlugin") as! AWSDataStorePlugin
     }
+    
+    func createErrorHandler(args: [String: Any])  -> DataStoreErrorHandler {
+        var errorHandler: DataStoreErrorHandler
+        if((args["hasErrorHandler"] as? Bool) == true) {
+            errorHandler = { error in
+                let map : [String:Any] = [
+                    "errorCode" : "DataStoreException",
+                    "errorMesage" : ErrorMessages.defaultFallbackErrorMessage,
+                    "details" : FlutterDataStoreErrorHandler.createSerializedError(error: error)
+                ]
+                self.channel!.invokeMethod("errorHandler", arguments: map)
+            }
+        } else {
+            errorHandler = { error in
+                Amplify.Logging.error(error: error)
+            }
+        }
+        return errorHandler
+    }
+    
+    func createConflictHandler(args: [String: Any])  -> DataStoreConflictHandler {
+        var conflictHandler:DataStoreConflictHandler
+        if((args["hasConflictHandler"] as? Bool) == true) {
+            conflictHandler = { conflictData, onDecision in
+                do {
+                    let modelName = conflictData.remote["__typename"] as! String
```

Comment:
Question: is the `conflictData.remote` from AppSync directly without native model conversion always where `__typename` is always available?

Reply:
If I understand you correctly, yes -> the __typename field should always be there in `conflictData.remote`.  Had to use that field due to an Amplify iOS quirk where fields labeled modelName were always returning `FlutterSerializedModel` which would break the code with "unknown model"